### ISSUE TITLE
CrossValidator puts telescope events from the same shower in different folds, so CV metrics are optimistic

### ISSUE THREAD

**Issue by vuillaut** [cta-observatory/ctapipe#3108]
The training tools run a k-fold cross-validation through CrossValidator.

But the data in the cross-validation sets are not independant from the training ones for two reasons:
- stereo events : One array event gives one row per triggered telescope of the given type. The same event seen by Tel A and B goes into training and validation
- shower reuse: as showers are reused 10 times in MC productions, correlated events go into training and validation

So the CV metrics are too optimistic. This would also affect hyperparameters tuning.
The IRFs and other metrics from the test set are not affected.

This could be fixed during the CV split.

**Comment by maxnoe** [comment]
> stereo events : One array event gives one row per triggered telescope of the given type. The same event seen by Tel A and B goes into training and validation

But wouldn't keeping them together also inflate the score? They are correlated. Correlation within the training or within the valdiation set would result also in biased estimates, not just correlation between the validation and the training set.

Keeping array events (or even re-used array events) in the same splits does not seem to me like it would yield more realistic estimates.
